### ISSUE TITLE
What is the recommended way to handle login banners?

### ISSUE THREAD

**Issue by scottchiefbaker** [salva/p5-Net-OpenSSH#23]
I'm using `Net::OpenSSH` to login to some Cisco equipment that has a login banner. The login banner spits back on `STDERR` which causes it to print to the screen. See raw shell example below:

```
nagios@green(~)
:ssh nagios@router.domain.com "show version" > /dev/null
User Access Verification
```

Note the "User Access Verification" which is the login banner. The only good way I've found to work around this is to use `Capture::Tiny` and wrapper my `new()` calls.

```
# The Nexus's have a login banner that can't be shutoff
# The banner is returned on STDERR before the login prompt
# so we have to capture it and ignore it
my $std_err = capture_stderr {
    $ssh = Net::OpenSSH->new($host,%$opts);
};

if ($ssh->error) {
    die("Couldn't establish SSH connection: ". $ssh->error);
}
```

If it were a command I could use `capture2()` and just throw away the `$err` but since it's a login banner on the `new()` I don't know the best way to handle this. I didn't see anything obvious in the documentation about login banners. There are a lot of options I can send `new()` but nothing seemed appropriate for what my needs.

How should I be handling login banners with `Net::OpenSSH`?

**Comment by salva** [comment]
You can redirect the `stderr` stream of the `ssh` master process to a file using the `master_stderr_fh`constructor option, or you can just discard it using `master_stderr_discard`.

**Comment by scottchiefbaker** [comment]
Doesnt't that redirect all of STDERR, even later `capture()` calls?

**Comment by salva** [comment]
No, every command has its own `stderr` stream. That option redirects the one coming from the master `ssh` process (usually the banner and errors related to connection issues),
